.github/workflows/build-deploy.yaml

```diff
@@ -20,27 +27,39 @@ jobs:
       repository: container-builder-template     
 
     strategy:
-
       # A matrix of Dockerfile paths and associated tags
       # Dockerfile in root builds to tag latest
       matrix:
         dockerfile: [[Dockerfile, latest], [subfolder/Dockerfile, subfolder]]
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Build ${{ matrix.dockerfile[1] }}
+
+      - name: Set Container Tag Normal
         run: |
           container="${{ env.registry }}/${{ env.username}}/${{ env.repository }}:${{ matrix.dockerfile[1]}}"
-          printf "Preparing to build ${container} from ${{ matrix.dockerfile[0] }}"
+          echo "container=${container}" >> $GITHUB_ENV
+
+      #   # On a new release create a container with the same tag as the release.
```

Comment:
Since this is a matrix, I think this would be potentially pushing multiple images with that same release tag (which could be different!) Should we perhaps include the tag and the GitHub ref here to be safe? If we do want a strict release tag version, we could check for the length of the Dockerfiles, and only do that if there is just one. What do you think?

Reply:
The other concern is that we might still want to deploy those named tags even if it's a release. So another idea would be to save an environment variable that is an array, and then loop through it so multiple tags are pushed for the same container.

Reply:
Oh that's a good point. I guess I assumed the sub-Dockerfile had a different name but you're right they're the same so it would be producing two.

Reply:
what if we just do the GitHub ref + tag? E.g., `0.0.1-latest`

Reply:
Great idea! Just pushed it!

Reply:
Perfect!